includes/frontend/class-wcal-checkout-process.php

```diff
@@ -551,7 +599,12 @@ public function wcal_order_placed( $order_id ) {
 				}
 			}
 
-			add_post_meta( $order_id, 'wcal_abandoned_cart_id', $abandoned_order_id_to_save );
+			if ( is_hpos_enabled() ) {
+				$order->add_meta_data( 'wcal_abandoned_cart_id', $abandoned_order_id_to_save );
+				$order->save();
```

Comment:
Instead of calling $order->save() each and every time we add/delete/update order meta, please call it once at the end of each function. This needs to be rectified in all the functions in this file.